client_test/directfund_test.go

```diff
@@ -75,73 +81,85 @@ func TestWhenObjectiveIsRejected(t *testing.T) {
+	return func(t *testing.T) {
 
-		// each client fetches the ConsensusChannel by reference to their counterparty
-		if store.GetChannelSecretKey() == &alice.PrivateKey {
-			con, ok = store.GetConsensusChannel(*clientB.Address)
-		} else {
-			con, ok = store.GetConsensusChannel(*clientA.Address)
+		// Spawn a pair of chain services
+		chainA, err := chainservice.NewSimulatedBackendChainService(sim, bindings, ethAccounts[0], logDestination)
+		if err != nil {
+			t.Fatal(err)
 		}
-
-		if !ok {
-			t.Fatalf("expected a consensus channel to have been created")
+		chainB, err := chainservice.NewSimulatedBackendChainService(sim, bindings, ethAccounts[1], logDestination)
+		if err != nil {
+			t.Fatal(err)
 		}
-		vars := con.ConsensusVars()
-		got := vars.Outcome.AsOutcome()
 
-		if diff := cmp.Diff(want, got); diff != "" {
-			t.Fatalf("expected outcome to be %v, got %v:\n %v", want, got, diff)
-		}
-		if vars.TurnNum != 1 {
-			t.Fatal("expected consensus turn number to be the post fund setup 1, received #$v", vars.TurnNum)
-		}
-		if con.Leader() != *clientA.Address {
-			t.Fatalf("Expected %v as leader, but got %v", clientA.Address, con.Leader())
+		broker := messageservice.NewBroker()
+
+		clientA, storeA := setupClient(alice.PrivateKey, chainA, broker, logDestination, 0)
+		clientB, storeB := setupClient(bob.PrivateKey, chainB, broker, logDestination, 0)
+
+		directlyFundALedgerChannel(t, clientA, clientB, asset)
+		want := testdata.Outcomes.Create(*clientA.Address, *clientB.Address, ledgerChannelDeposit, ledgerChannelDeposit, asset)
+
+		assertLedgerChannelInBothStoresWithOutcome := func(want outcome.Exit, storeA, storeB store.Store) {
+			for _, store := range []store.Store{storeA, storeB} {
+				var con *consensus_channel.ConsensusChannel
+				var ok bool
+
+				// each client fetches the ConsensusChannel by reference to their counterparty
+				if store.GetChannelSecretKey() == &alice.PrivateKey {
+					con, ok = store.GetConsensusChannel(*storeB.GetAddress())
+				} else {
+					con, ok = store.GetConsensusChannel(*storeA.GetAddress())
+				}
+
+				if !ok {
+					t.Fatalf("expected a consensus channel to have been created")
+				}
+				vars := con.ConsensusVars()
+				got := vars.Outcome.AsOutcome()
+
+				if diff := cmp.Diff(want, got); diff != "" {
+					t.Fatalf("expected outcome to be %v, got %v:\n %v", want, got, diff)
+				}
+				if vars.TurnNum != 1 {
+					t.Fatal("expected consensus turn number to be the post fund setup 1, received #$v", vars.TurnNum)
+				}
+				if con.Leader() != *storeA.GetAddress() {
+					t.Fatalf("Expected %v as leader, but got %v", *storeA.GetAddress(), con.Leader())
+				}
+
+				if !con.OnChainFunding.IsNonZero() {
+					t.Fatal("Expected nonzero on chain funding, but got zero")
+				}
+
+				if _, channelStillInStore := store.GetChannelById(con.Id); channelStillInStore {
+					t.Fatalf("Expected channel to have been destroyed in %v's store, but it was not", store.GetAddress())
+				}
+			}
 		}
```

Comment:
Defining this in-line here feels a little off to me - if it's meant for re-use, should it be a package function? If not, are there benefits to the encapsulation as a function?

Reply:
Good point. I think this is an artifact of a blind-alley I went up trying to extend the test. I think I will just flatten it back out for now, since it is a little too specific for re-use.

Reply:
[dda2478](https://github.com/statechannels/go-nitro/pull/1011/commits/dda2478eeeabec5e88ab48b250275902b52838fb)